lib/msf/core/handler/reverse_tcp.rb

```diff
@@ -89,7 +89,12 @@ def human_name
   def payload_uri
     addr = datastore['LHOST']
     uri_host = Rex::Socket.is_ipv6?(addr) ? "[#{addr}]" : addr
-    "tcp://#{uri_host}:#{datastore['LPORT']}"
+    if listener_sock.nil?
+      via = "(setting up)"
+    else
+      via = via_string(listener_sock.client) if listener_sock.respond_to?(:client)
+    end
+    "tcp://#{uri_host}:#{datastore['LPORT']} #{via}"
```

Comment:
Going to have a trailing space here if `via` is still `nil` after all these checks, might want to strip that out.